[Bug] Build fails when Automatic Mouse Layer is enabled

### Describe the Bug

When building with POINTING_DEVICE_AUTO_MOUSE_ENABLE specified, it fails to build and outputs the following error message:

```shell
$ Run qmk compile -j 4 -kb keyball/keyball44 -km default
Ψ Compiling keymap with gmake --jobs=[4](https://github.com/ikorihn/keyball/actions/runs/6361868099/job/17276562801#step:6:5) --output-sync=target keyball/keyball44:default

QMK Firmware 0.22.3
Making keyball/keyball44 with keymap default

Generating: .build/obj_keyball_keyball44_default/src/info_deps.d                                    [OK]
avr-gcc (GCC) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Generating: .build/obj_keyball_keyball44_default/src/default_keyboard.h                             [OK]
Generating: .build/obj_keyball_keyball44_default/src/default_keyboard.c                             [OK]
Generating: .build/obj_keyball_keyball44_default/src/info_config.h                                  [OK]
Compiling: keyboards/keyball/lib/oledkit/oledkit.c                                                  [OK]
Compiling: keyboards/keyball/keyball44/keyball44.c                                                  [OK]
Compiling: keyboards/keyball/drivers/pmw3360/pmw3360.c                                              [OK]
Compiling: keyboards/keyball/lib/keyball/keyball.c                                                  [OK]
Compiling: quantum/keymap_introspection.c                                                           [OK]
Compiling: .build/obj_keyball_keyball44_default/src/default_keyboard.c                              [OK]
Compiling: quantum/quantum.c                                                                        [OK]
Compiling: quantum/bitwise.c                                                                        [OK]
Compiling: quantum/led.c                                                                            [OK]
Compiling: quantum/action.c                                                                         [OK]
Compiling: quantum/action_tapping.c                                                                 [OK]
Compiling: quantum/action_layer.c                                                                   [OK]
Compiling: quantum/action_util.c                                                                    [OK]
Compiling: quantum/eeconfig.c                                                                       [OK]
Compiling: quantum/keymap_common.c                                                                  [OK]
Compiling: quantum/keyboard.c                                                                       [OK]
Compiling: quantum/keycode_config.c                                                                 [OK]
Compiling: quantum/sync_timer.c                                                                     [OK]
Compiling: quantum/logging/debug.c                                                                  [OK]
Compiling: quantum/logging/sendchar.c                                                               [OK]
Compiling: quantum/bootmagic/bootmagic_lite.c                                                       [OK]
Compiling: quantum/bootmagic/magic.c                                                                [OK]
Compiling: quantum/matrix_common.c                                                                  [OK]
Compiling: quantum/debounce/sym_defer_g.c                                                           [OK]
Compiling: quantum/matrix.c                                                                         [OK]
Compiling: quantum/split_common/transport.c                                                         [OK]
Compiling: quantum/split_common/split_util.c                                                        [OK]
Compiling: platforms/avr/printf.c                                                                   [OK]
Assembling: platforms/avr/xprintf.S                                                                 [OK]
Compiling: quantum/main.c                                                                           [OK]
Compiling: quantum/split_common/transactions.c                                                      [OK]
Compiling: quantum/pointing_device/pointing_device_drivers.c                                        [OK]
Compiling: quantum/pointing_device/pointing_device.c                                                [OK]
Compiling: quantum/via.c                                                                            [OK]
quantum/pointing_device/pointing_device_auto_mouse.c: In function 'auto_mouse_reset':
quantum/pointing_device/pointing_device_auto_mouse.c:107:[5](https://github.com/ikorihn/keyball/actions/runs/6361868099/job/17276562801#step:6:6): error: implicit declaration of function 'memset' [-Werror=implicit-function-declaration]
Compiling: quantum/pointing_device/pointing_device_auto_mouse.c                                     [ERRORS]
     memset(&auto_mouse_context.status, 0, sizeof(auto_mouse_context.status));
     ^~~~~~
quantum/pointing_device/pointing_device_auto_mouse.c:107:5: error: incompatible implicit declaration of built-in function 'memset' [-Werror]
quantum/pointing_device/pointing_device_auto_mouse.c:107:5: note: include '<string.h>' or provide a declaration of 'memset'
quantum/pointing_device/pointing_device_auto_mouse.c:24:1:
+#include <string.h>
 
quantum/pointing_device/pointing_device_auto_mouse.c:107:5:
     memset(&auto_mouse_context.status, 0, sizeof(auto_mouse_context.status));
     ^~~~~~
cc1: all warnings being treated as errors
gmake[1]: *** [builddefs/common_rules.mk:3[6](https://github.com/ikorihn/keyball/actions/runs/6361868099/job/17276562801#step:6:7)1: .build/obj_keyball_keyball44_default/quantum/pointing_device/pointing_device_auto_mouse.o] Error 1
gmake[1]: *** Waiting for unfinished jobs....
 | 
 | 
 | 
Compiling: quantum/crc.c                                                                            [OK]
Compiling: quantum/send_string/send_string.c                                                        [OK]
Compiling: drivers/oled/oled_driver.c                                                               [OK]
gmake: *** [Makefile:392: keyball/keyball44:default] Error 1
Make finished with errors

Error: Process completed with exit code 2.
```

### Keyboard Used

keyball/keyball44:default

### Link to product page (if applicable)

https://github.com/ikorihn/keyball (forked from https://github.com/Yowkees/keyball)

### Operating System

macOS

### qmk doctor Output

```text
❯ qmk doctor       
Ψ QMK Doctor is checking your environment.
Ψ CLI version: 1.1.2
Ψ QMK home: /Users/hiroki/repos/github.com/qmk/qmk_firmware
Ψ Detected macOS 14.0 (Apple Silicon).
Ψ Git branch: master
Ψ Repo version: 0.22.10
⚠ The official repository does not seem to be configured as git remote "upstream".
Ψ CLI installed in virtualenv.
Ψ All dependencies are installed.
Ψ Found arm-none-eabi-gcc version 13.2.0
Ψ Found avr-gcc version 9.4.0
⚠ We do not recommend avr-gcc newer than 8. Downgrading to 8.x is recommended.
Ψ Found avrdude version 7.2
Ψ Found dfu-programmer version 1.1.0
Ψ Found dfu-util version 0.11
Ψ Submodules are up to date.
Ψ Submodule status:
Ψ - lib/chibios: 2023-04-15 13:48:04 +0000 --  (11edb1610)
Ψ - lib/chibios-contrib: 2023-07-17 11:39:05 +0200 --  (da78eb37)
Ψ - lib/googletest: 2021-06-11 06:37:43 -0700 --  (e2239ee6)
Ψ - lib/lufa: 2022-08-26 12:09:55 +1000 --  (549b97320)
Ψ - lib/vusb: 2022-06-13 09:18:17 +1000 --  (819dbc1)
Ψ - lib/printf: 2022-06-29 23:59:58 +0300 --  (c2e3b4e)
Ψ - lib/pico-sdk: 2023-02-12 20:19:37 +0100 --  (a3398d8)
Ψ - lib/lvgl: 2022-04-11 04:44:53 -0600 --  (e19410f8)
Ψ QMK is ready to go, but minor problems were found
```


### Is AutoHotKey / Karabiner installed

- [ ] AutoHotKey (Windows)
- [ ] Karabiner (macOS)

### Other keyboard-related software installed

_No response_

### Additional Context

_No response_

--- a/quantum/pointing_device/pointing_device_auto_mouse.c
+++ b/quantum/pointing_device/pointing_device_auto_mouse.c
@@ -17,6 +17,7 @@
 
 #ifdef POINTING_DEVICE_AUTO_MOUSE_ENABLE
 
+#    include <string.h>
 #    include "pointing_device_auto_mouse.h"
 #    include "debug.h"
 #    include "action_util.h"

